### ISSUE TITLE
CorbTask does not seem to handle the xcc.httpcompliant flag

### ISSUE THREAD

**Issue by tzejohn** [rjrudin/ml-gradle#146]
When using the CorbTask to run a process against a host which is an AWS ELB, I am seeing BAD REQUEST errors. 

```
com.marklogic.xcc.exceptions.ServerResponseException: Query evaluation request rejected (400, BAD_REQUEST). Is this an XDBC server?
 [Session: user=admin, cb={default} [ContentSource: user=admin, cb={none} [provider: address=internal-ml-cluster-irx-dev-lb-1031654958.us-east-1.elb.amazonaws.com/10.239.12.224:8050, pool=0/64]]]
 [Client: XCC/8.0-5]
	at com.marklogic.xcc.impl.handlers.NotFoundCodeHandler.handleResponse(NotFoundCodeHandler.java:39)
	at com.marklogic.xcc.impl.handlers.EvalRequestController.serverDialog(EvalRequestController.java:96)
	at com.marklogic.xcc.impl.handlers.AbstractRequestController.runRequest(AbstractRequestController.java:88)
	at com.marklogic.xcc.impl.SessionImpl.submitRequestInternal(SessionImpl.java:437)
	at com.marklogic.xcc.impl.SessionImpl.submitRequest(SessionImpl.java:432)
	at com.marklogic.developer.corb.Manager.registerStatusInfo(Manager.java:609)
	at com.marklogic.developer.corb.Manager.init(Manager.java:184)
	at com.marklogic.developer.corb.AbstractManager.init(AbstractManager.java:175)
	at com.marklogic.developer.corb.Manager.main(Manager.java:135)
```

I did attempt to troubleshoot by executing the `com.marklogic.developer.corb.Manager` process directly, and I can see the same issue if I do not pass the `httpcompliant` flag. But if I pass it, then everything works. From mlGradle, it is not working, no matter if the flag is specified or not.

I am calling mlGradle with the VM argument `-Dxcc.httpcompliant=true`

**Comment by tzejohn** [comment]
From @hansenmc on the corb2 mailing list:

> The ml-gradle corb task performs a java exec, and currently only collects and propagates System properties that appear to be CoRB options. The task does not propagate the xcc.httpCompliant system property when it invokes CoRB.
> 
> The ml-gradle CorbTask collectSystemProperties() method:
> https://github.com/rjrudin/ml-gradle/blob/master/src/main/groovy/com/marklogic/gradle/task/CorbTask.groovy#L168
> 
> could be changed to include the xcc.httpCompliant property:
> 
>   public Map collectSystemProperties() {
>     System.properties.findAll { corbOptions.containsValue(it.key) || it.key.equals(“xcc.httpCompilant”) }
>   }
> 
> or even adjusted to collect all of the System.properties without filtering. That would be the easiest, but would run the risk of some system properties from the ml-gradle instance being applied to a CoRB job that were not intended.
> 

@rjrudin what is you preference for the change, just this additional option to pass through, or all options?

**Comment by rjrudin** [comment]
Are there any others that make sense to pass through? If not, I say just support xcc.httpCompliant for now.

**Comment by rjrudin** [comment]
@hansenmc I can make this fix if you can show me how to do it (I'm not familiar enough with CorbTask yet). I can always toss it into a bugfix release - about to make 2.5.0.

**Comment by hansenmc** [comment]
I can make the change and submit a PR. I was leaning toward passing any/all System properties, instead of only the xcc.httpCompliant - since it would provide a means of specifying custom input properties with this task and resolve issue #128. 

If we want to be more judicious, then at least include a pattern looking for flags that start with URIS-MODULE.*, PROCESS-MODULE.*, etc.

**Comment by rjrudin** [comment]
I like the idea of passing all System properties, I can't think of a reason why that would cause something to break.

**Comment by rjrudin** [comment]
I believe this can be closed as a result of the PR - is available in ml-gradle 2.5.0, which I'm preparing now.

**Comment by hansenmc** [comment]
FYI - I created a [CoRB2 issue](https://github.com/marklogic/corb2/issues/45) to provide the ability to set an option for setting "http.compliant" flag that when specified, CoRB will set the system property. So, starting in the next release, it will be possible to set that option in the options file and not **_have to_** set it as a system property when invoking CoRB.
